src/login/logind-session.c

```diff
@@ -304,6 +318,33 @@ int session_save(Session *s) {
         return log_error_errno(r, "Failed to save session data %s: %m", s->state_file);
 }
 
+static int session_load_devices(Session *s, const char *devices) {
+        _cleanup_free_ char *word = NULL;
+        const char *p;
+        int r;
+
+        assert(s);
+
+        for (p = devices;;) {
+                SessionDevice *sd;
+                dev_t dev;
+
+                r = extract_first_word(&p, &word, NULL, 0);
+                if (r <= 0)
+                        break;
+
+                r = parse_dev(word, &dev);
```

Comment:
word is leaked. The declaration with cleanup must be inside of the loop.

I think it would be prudent to emit something on error, maybe at debug level.